plugins/contentstack/src/plugin.ts

```diff
@@ -112,24 +111,62 @@ registerDataPlugin(
             id: model.uid,
             description: model.description,
             inputs: () => {
-              return [
+              const subFields = model.schema
+                // searching by ref UID seems pointless, so we exclude that for now.
+                // In the future, we could add searching within references:
+                // https://www.contentstack.com/docs/developers/apis/content-delivery-api/#reference-search-equals
+                .filter(field => field.data_type !== 'reference')
+                .map(
+                  (field): Input => ({
+                    name: field.uid,
+                    friendlyName: field.display_name,
+                    type: field.data_type,
+                  })
+                );
+
+              const inputs: Input[] = [
```

Comment:
This big block could probably be moved to function like `buildInputs` in the same module.

Reply:
Done in 27e07d1